frontend/src/app/messages/messages.service.spec.ts

```diff
@@ -0,0 +1,171 @@
+import { fakeAsync, tick, TestBed, inject } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { Observable } from 'rxjs/Observable';
+import { ISubscription } from 'rxjs/Subscription';
+import { environment } from '../../environments/environment';
+
+
+import {
+  MessagesService, Message,
+  MessageType, DEFAULT_TIMEOUT_TIME, ServerMessage
+} from './messages.service';
+
+describe('MessagesService', () => {
+  let service: MessagesService;
+  let httpMock: HttpTestingController;
+  let testMessage: Message;
+  let numberOfCalls: number;
+  let messageSubscription: ISubscription;
+
+  this.subscribeToMessage$ = () => {
+    messageSubscription = service.message$.subscribe(msg => {
+      expect(msg).toEqual(testMessage)
+      numberOfCalls++;
+    })
+  }
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [MessagesService]
+    });
+
+    service = TestBed.get(MessagesService);
+    httpMock = TestBed.get(HttpTestingController);
+    testMessage = {
+      message: 'Test message',
+      type: MessageType.Info,
+      dismissible: true,
+      timeout: DEFAULT_TIMEOUT_TIME
+    }
+    numberOfCalls = 0;
```

Comment:
I'm sure that jasmine spies have callCount and there is not need for make shift solution like that

Reply:
Done